src/utils.ts

```diff
@@ -1,11 +1,80 @@
 import { constants } from './prelude';
-import { ethers } from 'hardhat';
+import hre, { ethers } from 'hardhat';
 import { time } from '@nomicfoundation/hardhat-network-helpers';
-import { providers, Wallet, Contract, Bytes, ContractTransaction, BigNumberish } from 'ethers';
+import { providers, Wallet, Contract, Bytes, ContractTransaction, BigNumberish, BigNumber } from 'ethers';
+import { DeployOptions, DeployResult } from 'hardhat-deploy/types';
+
+const _delay = function (ms: number) {
+    new Promise((resolve) => {
+        setTimeout(resolve, ms);
+    });
+};
```

Comment:
```suggestion
function _delay (ms: number) {
    return new Promise(resolve => setTimeout(resolve, ms));
}
```

Reply:
Currently it's broken, because doesn't return promise

Reply:
Thanks for pointing this out.

I'm now wondering about one thing. Is this function needed in the library and if so, should it be exported? It originally appeared in this PR because it was used in one of new functions. It seems to be possible to do without it in this file now, as it is only used once. What do you think, is it worth keeping it and using it in projects or is the functionality simple enough and one doesn't have to bother importing it every time?

Reply:
Not sure exporting it makes any sense, too primitive trick and probably not a good practice :))

Reply:
Agreed, removed the `_delay` function at all and returned it to the `timeIncreaseTo` function to use `Promise` directly.